src/main/java/org/jenkinsci/plugins/casc/ConfigurationAsCode.java

```diff
@@ -136,24 +143,67 @@ public static void init() throws Exception {
     }
 
     /**
-     * Main entry point to start configuration process
+     * Main entry point to start configuration process.
      * @throws ConfiguratorException Configuration error
      */
     public void configure() throws ConfiguratorException {
+        List<String> configParameters = getBundledCasCURIs();
+        if (!configParameters.isEmpty()) {
+            LOGGER.log(Level.FINE, "Located bundled config YAMLs: {0}", configParameters);
+        }
+
         String configParameter = System.getProperty(
                 CASC_JENKINS_CONFIG_PROPERTY,
                 System.getenv(CASC_JENKINS_CONFIG_ENV)
         );
         if (configParameter == null) {
             if (Files.exists(Paths.get(DEFAULT_JENKINS_YAML_PATH))) {
                 configParameter = DEFAULT_JENKINS_YAML_PATH;
-            } else {
-                // No configuration set nor default config file
-                return;
             }
         }
 
-        configure(configParameter);
+        if (configParameter != null) {
+            // Add external config parameter
+            configParameters.add(configParameter);
+        }
+        if (configParameters.isEmpty()) {
+            LOGGER.log(Level.FINE, "No configuration set nor default config file");
+            return;
+        }
+        configure(configParameters);
+    }
+
+    public List<String> getBundledCasCURIs() {
+        final String cascFile = "casc.yml";
+        final String cascDirectory = "casc.yml.d";
+        List<String> res = new ArrayList<>();
+
+        final ServletContext servletContext = Jenkins.getInstance().servletContext;
+        try {
+            URL bundled = servletContext.getResource("/WEB-INF/"+ cascFile);
+            if (bundled != null) {
+                res.add(bundled.toString());
+            }
+        } catch (IOException e) {
+            LOGGER.log(WARNING, "Failed to load /WEB-INF/" + cascFile, e);
+        }
+
+        Set<String> resources = servletContext.getResourcePaths("/WEB-INF/"+ cascDirectory +"/");
```

Comment:
Would it make sense to only look for `.yml` or `.yaml` files here? Are there any situations where people might end up putting non yaml files in this location?

Reply:
Agree, should filter by YAML_FILES_PATTERN as we do for a fielsystem-based conf.d

Reply:
Agreed